doc/api/esm.md

```diff
@@ -1286,28 +1284,25 @@ unknown-to-Node.js file extensions. See the [transpiler loader example][] below.
 
 ```js
 /**
- * @param {string|buffer} source
- * @param {object} context
- * @param {string} context.url
- * @param {string} context.format
- * @param {function} defaultTransformSource
- * @returns {object} response
- * @returns {string|buffer} response.source
+ * @param {(SharedArrayBuffer | string | Uint8Array)} source
```

Comment:
Can it not also be just `Buffer`?

I had a discussion with @bmeck about this in the original PR where this was written.

Reply:
Would you mind linking the thread?

Reply:
Not to rush the process, but I will have to proceed w/o the response to that question.

> Can it not also be just `Buffer`?

1. [“These types all correspond to classes defined in ECMAScript.”](https://github.com/nodejs/node/blame/47044a91c6269a395098cf74d4b2a2e70df5c321/doc/api/esm.md#L1211)
2. [`Buffer`](https://github.com/microsoft/TypeScript/blob/9119fe3797d4c7dd2b648fc6d34633cee9a2653f/src/compiler/sys.ts#L1025) is not an ECMAScript type.
3. [Closure Compiler warns about *type misinformation*.](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540warning_level%2520VERBOSE%250A%252F%252F%2520%2540jscomp_error%2520strictCheckTypes%250A%252F%252F%2520%2540language_out%2520ECMASCRIPT_NEXT%250A%252F%252F%2520%2540checks_only%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**%250A%2520*%2520%2540param%2520%257Bstring%257D%2520url%250A%2520*%2520%2540param%2520%257B%257B%2520format%253A%2520string%2520%257D%257D%2520context%250A%2520*%2520%2540param%2520%257BFunction%257D%2520defaultGetSource%250A%2520*%2520%2540returns%2520%257BPromise%253C%257B%2520source%253A%2520(Buffer%2520%257C%2520string)%2520%257D%253E%257D%250A%2520*%252F%250Aexport%2520async%2520function%2520getSource(url%252C%2520context%252C%2520defaultGetSource)%2520%257B%250A%2520%2520const%2520%257B%2520format%2520%257D%2520%253D%2520context%253B%250A%2520%2520if%2520(Math.random()%2520%253E%25200.5)%2520%257B%2520%252F%252F%2520Some%2520condition.%250A%2520%2520%2520%2520%252F%252F%2520For%2520some%2520or%2520all%2520URLs%252C%2520do%2520some%2520custom%2520logic%2520for%2520retrieving%2520the%2520source.%250A%2520%2520%2520%2520%252F%252F%2520Always%2520return%2520an%2520object%2520of%2520the%2520form%2520%257Bsource%253A%2520%253Cstring%257Cbuffer%253E%257D.%250A%2520%2520%2520%2520return%2520%257B%250A%2520%2520%2520%2520%2520%2520source%253A%2520'...'%252C%250A%2520%2520%2520%2520%257D%253B%250A%2520%2520%257D%250A%2520%2520%252F%252F%2520Defer%2520to%2520Node.js%2520for%2520all%2520other%2520URLs.%250A%2520%2520return%2520defaultGetSource(url%252C%2520context%252C%2520defaultGetSource)%253B%250A%257D)
3. *Type misinformation* results in issues like https://github.com/nodejs/node/issues/33585.

So, probably not?

Reply:
https://github.com/nodejs/node/pull/32202#issuecomment-602634577

Reply:
@GeoffreyBooth, sorry for the delay on this. I guess the question has to do w/ the following. https://github.com/nodejs/node/blob/641c466671556b179f1cdb7f9cbb36f4750c253d/doc/api/buffer.md#L12-L15

Take a look at the following test.

https://github.com/nodejs/node/pull/32202/files#diff-6fbec65c7e96fe3dd9089e93bf9444f1R1-R50

Nowhere is `Buffer` an acceptable type. What can I do to resolve this conversation?

Reply:
It's not a block, I just wanted you to check that this change was correct 😀 If you're sure that the new type is how it should be, that's good enough for me.

Reply:
I just noticed that this should not accept null.

https://github.com/nodejs/node/pull/32202/files#diff-6fbec65c7e96fe3dd9089e93bf9444f1R16

I will have to rebase to ensure that the annotation excludes that. Will also revise that there aren't other problems related to accepting null when we shouldn't.

Reply:
Alright, so now `null` has been negated from these type sets. Just as a reminder, we just want to do the minimal amount of negation to keep the compilers quiet while also staying true to the tests. TypeScript is not set to `strictNullChecks` because that is not the default behavior and Closure is doing this job for us anyways. So we should be good to go here.

Reply:
☝️ My initial thought was to negate `null` in all the places where it should not appear, but then there is an argument that we do not want exclamation points everywhere.

Nobody has objected to providing copy-pasteable code snippets for the hooks and I still think it's a good idea for the sake of convenience and education. TypeScript's docs have JSDoc snippets, so I would value the opinion of @weswigham on whether this seems like an appropriate philosophy for snippets. Any further comments would be appreciated.